cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -376,6 +379,9 @@ public OptimizerResult optimizations(ClusterModel clusterModel,
    * @param isTriggeredByGoalViolation True if optimization of goals is triggered by goal violation, false otherwise.
    * @param requestedDestinationBrokerIds Explicitly requested destination broker Ids to limit the replica movement to
    *                                      these brokers (if empty, no explicit filter is enforced -- cannot be null).
+   * @param initReplicaDistributionForProposalGeneration The initial replica distribution of the cluster.
+   * @param initLeaderDistributionForProposalGeneration The initial leader replica distribution of the cluster.
+   * @param onlyMoveImmigrantReplicas Whether restrict replica movement only to immigrant replicas or not.
```

Comment:
Do we need `onlyMoveImmigrantReplicas`?
Shouldn't it be inferred from the fact that wither `initReplicaDistributionForProposalGeneration != null` or `initLeaderDistributionForProposalGeneration != null`? Are there cases that conflict with this assumption?

Reply:
Currently the assumption hold, but in the future we may have another scenario that needs to specify `onlyMoveImmigrantReplicas` to be `true`.
Fundamentally since `initReplicaDistributionForProposalGeneration`/`initLeaderDistributionForProposalGeneration` and `onlyMoveImmigrantReplicas` serving two different purpose(one is for proposal diff, one is an requirement for proposal calculation), I prefer not tightly coupled them.